### PR TITLE
Mob x

### PR DESCRIPTION
**Pull request by RehanPaismo**

### CODE REVIEW

File: src/Components/AddToDo.jsx [hunk 1]
```diff
@@ -1,19 +1,25 @@
+import { observer } from "mobx-react";
 import React, { useState } from "react";
 import "./styles.css";
 
-const AddToDo = () => {
-  const [list, setList] = useState([]);
+const AddToDo = observer (({myProp}) => {
+  // const [list, setList] = useState([]);
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx]: always delete commented code

File: src/Components/AddToDo.jsx [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
   const [todo, setTodo] = useState();
 
   const add = () => {
     if (todo !== ""){
-    setList([...list, todo]);
+    // setList([...list, todo]);
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx]: same

File: src/Components/AddToDo.jsx [hunk 3]
```diff
@@ -30,14 +36,18 @@ const AddToDo = () => {
           <input type="submit" value="ADD" onClick={add}></input>
         
         </span>
+        <span>
+          <input type="submit" value="Clear" onClick={clearTodo}></input>
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx]:
```suggestion
          <input type="submit" value="Clear" onClick={clearTodo} />
```

File: src/Components/AddToDo.jsx [hunk 4]
```diff
@@ -30,14 +36,18 @@ const AddToDo = () => {
...
+        
+        </span>
       </div>
       <div className="list">
-        {list.map((element, index) => {
-          return <div className="items" key={index}>{element}</div>;
+        {myProp.todoArray.map((element) => {
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx]: what if `myProp.todoArray` is undefined? in that case it will crash

File: src/Components/TodoClass.js [hunk 5]
```diff
@@ -1,12 +1,22 @@
+import {action, makeObservable, observable} from 'mobx'
 class ToDoClass {
    
     todoArray = [];
+    constructor(){
+        makeObservable(this, {
+            todoArray:observable,
```
Comment (saqlain-paismo) [inline src/Components/TodoClass.js]: `todoArray` is not a appropriate name  avoid naming things as class, array or object. the name should automatically suggests its an array.
i.e. in this case you could have used `todos`

File: src/Components/TodoClass.js [hunk 6]
```diff
@@ -1,12 +1,22 @@
+import {action, makeObservable, observable} from 'mobx'
 class ToDoClass {
```
Comment (saqlain-paismo) [inline src/Components/TodoClass.js]: same a avoid `class` in this name, and class are always going to be singular.

File: src/Components/TodoClass.js [hunk 7]
```diff
@@ -1,12 +1,22 @@
...
    
     todoArray = [];
+    constructor(){
+        makeObservable(this, {
+            todoArray:observable,
+            addToDo:action
+        })
+    }
     addToDo = (todoInput) => {
         const todoObj ={
```
Comment (saqlain-paismo) [inline src/Components/TodoClass.js]: same avoid data type names while naming things

File: src/Components/AddToDo.jsx [hunk 8]
```diff
@@ -1,19 +1,25 @@
+import { observer } from "mobx-react";
 import React, { useState } from "react";
 import "./styles.css";
 
-const AddToDo = () => {
-  const [list, setList] = useState([]);
+const AddToDo = observer (({myProp}) => {
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx]: `myProp` is not a good name always say `props`

File: src/Components/AddToDo.jsx [hunk 9]
```diff
@@ -1,19 +1,25 @@
...
+  // const [list, setList] = useState([]);
   const [todo, setTodo] = useState();
 
   const add = () => {
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx]: add what?

**Review by saqlain-paismo — CHANGES_REQUESTED** [review]

File: src/Components/AddToDo.jsx (hunk 1 above)
Comment (RehanPaismo) [reply src/Components/AddToDo.jsx]: Understand

File: src/Components/TodoClass.js (hunk 5 above)
Comment (RehanPaismo) [reply src/Components/TodoClass.js]: Ok, Understand

File: src/Components/TodoClass.js (hunk 6 above)
Comment (RehanPaismo) [reply src/Components/TodoClass.js]: Done and Understand

File: src/Components/AddToDo.jsx (hunk 9 above)
Comment (RehanPaismo) [reply src/Components/AddToDo.jsx]: This was an add todo function. changes add to addTodo

File: src/Components/AddToDo.jsx (hunk 8 above)
Comment (RehanPaismo) [reply src/Components/AddToDo.jsx]: Understand

File: src/Components/AddToDo.jsx (hunk 4 above)
Comment (RehanPaismo) [reply src/Components/AddToDo.jsx]: It's Done. Applied Check on this

File: src/App.jsx [hunk 10]
```diff
@@ -1,11 +1,11 @@
-
 import AddToDo from './Components/AddToDo'
+import {toDo} from './Components/TodoClass'
 function App() {
   
 
   return (
     <>
-      <AddToDo />
+      <AddToDo myProps={toDo}/>
```
Comment (saqlain-paismo) [inline src/App.jsx:8]: `props`

File: src/App.jsx [hunk 11]
```diff
@@ -1,11 +1,11 @@
-
 import AddToDo from './Components/AddToDo'
+import {toDo} from './Components/TodoClass'
```
Comment (saqlain-paismo) [inline src/App.jsx:2]: access through mobx hook `useMobxContext`

File: src/Components/AddToDo.jsx [hunk 12]
```diff
@@ -1,19 +1,31 @@
...
   };
 
   const updateTodo = (e) =>{
     setTodo(e.target.value);
   }
+
+  const clearTodo = () => {
+    myProps.clear();
+  }
+  const printToDoList = () =>{
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx:21]: `renderTodoList`

File: src/Components/AddToDo.jsx [hunk 13]
```diff
@@ -1,19 +1,31 @@
...
+    
+    if (myProps.todoItems.length !== 0){
+      return myProps.todoItems.map((element) => {
+        return <div className="items" key={element.id}>{element.item}</div>;
+      })
```
Comment (saqlain-paismo) [inline src/Components/AddToDo.jsx:26]:
```suggestion
      return myProps.todoItems.map((element) => (<div className="items" key={element.id}>{element.item}</div>)
      )
```

**Review by saqlain-paismo — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 10 above)
Comment (RehanPaismo) [reply src/App.jsx:8]: Fixed

File: src/Components/AddToDo.jsx (hunk 12 above)
Comment (RehanPaismo) [reply src/Components/AddToDo.jsx:21]: Fixed

File: src/Components/AddToDo.jsx (hunk 13 above)
Comment (RehanPaismo) [reply src/Components/AddToDo.jsx:26]: Follow Guide

File: src/App.jsx (hunk 11 above)
Comment (RehanPaismo) [reply src/App.jsx:2]: useMobxContexet is a custom hook

**Pull request merged**